### DISCUSSION TITLE
Reload edit or view page after an Action

### DISCUSSION THREAD

**Discussion by R00tes** [filamentphp/filament#1648] [General]
Hello, 
Sorry if is a noob question, I have a Modal action and I would like to refresh my Edit page after I complied that action.
For sure is really easy and I will look really bad.. hahaha

Thanks in advance.

**Comment by danharrin** [comment]
Hey! Try `$this->record->refresh()` maybe? Or do you mean reloading the form data? What happens if it's unsaved?

**Reply by R00tes** [reply]
Yes, reloading the form data, the data is saved but I have to reload the page to show the new data.

I have this in the EditRecord 

>    protected function getActions(): array

    {
        return [
            \Filament\Pages\Actions\ButtonAction::make(__('Terminating contract'))
                ->hidden(($this->record->notification_ending) ? true : false)
                ->requiresConfirmation()
                ->color('warning')
                ->icon('heroicon-s-ban')
                ->action(function (array $data): void {
                    $this->record->notification_ending = $data['notification_ending'];
                    $this->record->save();
                    $this->record->refresh();
                    $this->notify('warning', 'Notified Cancelation', isAfterRedirect: false);
                })
                ->form([
                    \Filament\Forms\Components\DatePicker::make('notification_ending')
                    ->label('Notification of cancelation at:')
                    ->format('d.m.Y')
                    ->displayFormat('d.m.Y')
                    ->required()
                ])
        ];
    }

**Reply by danharrin** [reply]
Try `$this->fillForm()`! It will refill the form will values from the database.

**Reply by R00tes** [reply]
Perfect! Thanks so much

**Reply by rvzug** [reply]
@danharrin is there a generic way to refresh a page (reload the form on EditRecord-page, but reload the record on a ViewRecord-page, reload the table on a ListRecords-page etc.)? So not depending on the type?

**Comment by RX-77IT** [comment]
To expand a little (and to remind myself in the future), there is also the `refreshFormData()` method to selectively refresh fields: [https://filamentphp.com/docs/3.x/panels/pages#refreshing-form-data](https://filamentphp.com/docs/3.x/panels/pages#refreshing-form-data)

In my case I had a Component Section with a headerAction and inside the modal action I didn't have access to `$this` (error "Using $this when not in object context"). I made the refreshFormData() work by injecting the current livewire component instance.

```php
use Filament\Forms;

// inside a form() of a Resource
Forms\Components\Section::make('Label')
    ->headerActions([
        Forms\Components\Actions\Action::make('action_name')
            // ...
            ->action(function (?Model $record, \Livewire\Component $livewire) {
                // modify $record fields
                $record->save();

                $livewire->refreshFormData([
                    'name_of_the_field_modified_1',
                    'name_of_the_field_modified_2',
                    // ...
                ]);
            }),
    ])
```

**Comment by rvzug** [comment]
For those who are struggeling with a (relation) repeater inside a modal-form, who want to display the repeater order:

```php
Repeater::make('list')
  ->afterStateUpdated(function (array $state, Livewire $livewire): void {
      // save the order manually (in my case)
      $livewire->getMountedTableActionForm()->fill([]);
  })
```
This refills the form. The empty array loads the saved order from the database.
